include/deal.II/multigrid/mg_data_out.h

```diff
@@ -0,0 +1,153 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mg_data_out_h
+#define dealii_mg_data_out_h
+
+#include <deal.II/base/mg_level_object.h>
+
+#include <deal.II/numerics/data_out.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup mg */
+/*@{*/
+
+/**
+ * Specialization of DataOut for output of all cells on a single level.
+ *
+ * This class allows for data output of all cells (active or not) on a given
+ * level. If the underlying Triangulation is parallel, only cells with the local
+ * level_subdomain_id are visualized.
+ *
+ * This class is particularly useful to output solutions on a
+ * specific multigrid level.
+ */
+template <int dim, typename DoFHandlerType = DoFHandler<dim>>
+class SingleLevelDataOut : public DataOut<dim, DoFHandlerType>
+{
+public:
+  /**
+   * Default constructor. Use reinit() to initialize the class.
+   */
+  SingleLevelDataOut()
+    : level(numbers::invalid_unsigned_int)
+  {}
+
+  /**
+   * Constructor.
+   * @param level Cells have to be on this level to be output.
+   */
+  explicit SingleLevelDataOut(const unsigned int level)
+    : level(level)
+  {}
+
+  /**
+   * Reset the options of this class (see the constructor above).
+   */
+  void
+  reinit(const unsigned int level)
+  {
+    this->level = level;
+  }
+
+  /**
+   * Implementation to get the first cell to output.
+   */
+  virtual typename DataOut<dim>::cell_iterator
```

Comment:
Here and below you're consistently missing the second template argument of the `DataOut` base class -- i.e., you're not referring to the correct type because it may not be the base class.